server/etcdserver/server.go

```diff
@@ -1833,6 +1836,18 @@ func (s *EtcdServer) apply(
 			s.setAppliedIndex(e.Index)
 			s.setTerm(e.Term)
 			shouldStop = shouldStop || removedSelf
+
+			// etcdserver need to ensure the raft has already been notified
+			// or advanced before it responds to the client. Otherwise, the
+			// following config change request may be rejected.
+			// See https://github.com/etcd-io/etcd/issues/15528.
+			select {
+			case <-time.After(maxRaftAdvanceDelay):
```

Comment:
Second thought, from linearizability perspective, we should not. It was carried over from the original commit and I was thinking member promote response can be still be sent out even with slow disk with fsync latency because it has already applied even with failure.